src/main/java/com/epam/ta/reportportal/dao/TestItemRepositoryCustomImpl.java

```diff
@@ -389,6 +391,39 @@ public TestItemTypeEnum getTypeByItemId(Long itemId) {
 		return PageableExecutionUtils.getPage(items, pageable, () -> dsl.fetchCount(QueryBuilder.newBuilder(filter).build()));
 	}
 
+	@Override
+	public List<NestedStep> findAllNestedStepsByIds(Collection<Long> ids) {
+
+		return dsl.select(TEST_ITEM.ITEM_ID,
+				TEST_ITEM.NAME,
+				TEST_ITEM.START_TIME,
+				TEST_ITEM.TYPE,
+				TEST_ITEM_RESULTS.STATUS,
+				TEST_ITEM_RESULTS.END_TIME,
+				TEST_ITEM_RESULTS.DURATION,
+				DSL.field(DSL.exists(dsl.selectOne().from(LOG).where(LOG.ITEM_ID.eq(TEST_ITEM.ITEM_ID)).limit(1))
+						.orExists(dsl.selectOne()
+								.from(TEST_ITEM.as(NESTED))
+								.where(fieldName(NESTED, TEST_ITEM.PARENT_ID.getName()).cast(Long.class).eq(TEST_ITEM.ITEM_ID))
```

Comment:
as i can see there are several places in the method where aliases is used
maybe it's a good idea to create an aliase in the beginning of the method to avoid fields casting?

Reply:
done